lib/Doctrine/DBAL/Connection.php

```diff
@@ -937,7 +944,7 @@ public function executeCacheQuery($query, $params, $types, QueryCacheProfile $qc
     {
         $resultCache = $qcp->getResultCacheDriver() ?: $this->_config->getResultCacheImpl();
         if ( ! $resultCache) {
```

Comment:
It would be simpler if enforced by the interfaces of `$qcp` and the config. E.g they both are required to have an instance of the cache driver which defaults to a null-implementation which throws exceptions from any method.

Reply:
moved to #3165